django_cloud_deploy/cli/prompt.py

```diff
@@ -295,239 +432,382 @@ def _generate_default_project_id(project_name=None):
+               args: Dict[str, Any]) -> Dict[str, Any]:
+        prompter = GoogleNewProjectId()
+
+        if args.get('use_existing_project', False):
+            prompter = GoogleExistingProjectId(self.project_client)
+
+        return prompter.prompt(console, step, args)
+
+
+class GoogleExistingProjectId(TemplatePrompt):
+    """Handles Project ID for existing projects."""
+
+    PARAMETER = 'project_id'
+
+    def __init__(self, project_client: project.ProjectClient):
+        self.project_client = project_client
+
+    def prompt(self, console: io.IO, step: str,
+               args: Dict[str, Any]) -> Dict[str, Any]:
         """Prompt the user to a Google Cloud Platform project id.
 
-        Args:
-            console: Object to use for user I/O.
-            step_prompt: A prefix showing the current step number e.g. "[1/3]".
-            arguments: The arguments that have already been collected from the
-                user e.g. {"project_id", "project-123"}
-            credentials: The OAuth2 Credentials object to use for api calls
-                during prompt.
+        If the user supplies the project_id as a flag we want to validate that
+        it exists. We tell the user to supply a new one if it does not.
 
-        Returns:
-            The value entered by the user.
         """
-        default_project_id = cls._generate_default_project_id(
-            arguments.get('project_name', None))
-        while True:
-            console.tell(('{} Enter a Google Cloud Platform Project ID, '
-                          'or leave blank to use').format(step_prompt))
-            project_id = console.ask('[{}]: '.format(default_project_id))
-            if not project_id.strip():
-                return default_project_id
-            try:
-                cls.validate(project_id)
-            except ValueError as e:
-                console.error(e)
-                continue
-            return project_id
 
-    @staticmethod
-    def validate(s, credentials: Optional[credentials.Credentials] = None):
+        new_args = copy.deepcopy(args)
+        if self._is_valid_passed_arg(console, step,
+                                     args.get(self.PARAMETER, None),
+                                     self._validate):
+            return new_args
+
+        msg = ('{} Enter the <b>existing<b> Google Cloud Platform Project ID '
+               'to use.').format(step)
+        answer = _ask_prompt(msg, console, self._validate)
+        new_args[self.PARAMETER] = answer
+        return new_args
+
+    def _validate(self, s: str):
         """Validates that a string is a valid project id.
 
         Args:
             s: The string to validate.
-            credentials: The OAuth2 Credentials object to use for api calls
-                during validation.
 
         Raises:
             ValueError: if the input string is not valid.
         """
+
         if not re.match(r'[a-z][a-z0-9\-]{5,29}', s):
```

Comment:
Why not ?

Reply:
True, but havent you ever forgotten a string and its because it has wierd things that you forget? If the user uses my_string theyll be reminded its my-string (hopefully)

Reply:
OK